docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc

```diff
@@ -0,0 +1,233 @@
+- `attributes.results.updated`: Anonymization field objects that were updated during the action's execution
+- `attributes.results.deleted`: Anonymization field objects that were deleted during the action's execution
+- `attributes.results.skipped`: Anonymization field that were skipped during the action's execution
+
+An anonymization field can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a filed to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include prompts' `id`, `name`, and `skip_reason`.
+
+[source,json]
+--------------------------------------------------
+{
+    "success": true,
+    "anonymization_fields_count": 2,
+    "attributes": {
+        "results": {
+            "updated": [],
+            "created": [
+                {
+                    "timestamp": "2024-08-13T20:54:23.125Z",
+                    "createdAt": "2024-08-13T20:54:23.125Z",
+                    "field": "agent.type",
+                    "allowed": true,
+                    "anonymized": false,
+                    "updatedAt": "2024-08-13T20:54:23.125Z",
+                    "id": "cB2FTZEBYaDeA-NhPHqW"
+                },
+                {
+                    "timestamp": "2024-08-13T20:54:23.125Z",
+                    "createdAt": "2024-08-13T20:54:23.125Z",
+                    "field": "agent.name",
+                    "allowed": true,
+                    "anonymized": true,
+                    "updatedAt": "2024-08-13T20:54:23.125Z",
+                    "id": "cR2FTZEBYaDeA-NhPHqW"
+                }
+            ],
+            "deleted": [],
+            "skipped": []
+        },
+        "summary": {
+            "failed": 0,
+            "succeeded": 2,
+            "skipped": 0,
+            "total": 2
+        }
+    }
+}
+--------------------------------------------------
+
+*Example 2, Partial failure*
+
+The following request adds tags `tag-1` and `tag-2` to the rules that have the IDs sent in the payload:
+[source,console]
+--------------------------------------------------
+POST api/security_ai_assistant/anonymization_fields/_bulk_action
+{
+  "delete": {
+    ids: [
+      "cR2FTZEBYaDeA-NhPHqW"
+    ]
+  },
+   "update":[
+      {
+         "id":"lh12SZEBYaDeA-NhmkwG",
+         "allowed":false
+      },
+      {
+         "id":"lR12SZEBYaDeA-NhmkwG",
+         "anonymized":true
+      }
+   ]
+}
+--------------------------------------------------
+
+[discrete]
+===== Response code
+
+`500`::
+    Indicates partial bulk action failure.
+
+[discrete]
+===== Response payload
+
+If processing of any rule fails, a partial error outputs the ID and/or name of the affected rule and the corresponding error, as well as successfully processed rules (in the same format as a successful 200 request).
```

Comment:
What does "rules" mean here? I thought this API was about updating field anonymization settings.

Reply:
copied from there